dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/deprecated/tracker/event/EventManagerTest.java

```diff
@@ -0,0 +1,101 @@
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.dxf2.deprecated.tracker.event;
+
+import static org.hisp.dhis.utils.Assertions.assertContainsOnly;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.hisp.dhis.dxf2.deprecated.tracker.event.persistence.EventPersistenceService;
+import org.hisp.dhis.dxf2.deprecated.tracker.importer.Checker;
+import org.hisp.dhis.dxf2.deprecated.tracker.importer.EventManager;
+import org.hisp.dhis.dxf2.deprecated.tracker.importer.EventProcessorExecutor;
+import org.hisp.dhis.dxf2.deprecated.tracker.importer.EventProcessorPhase;
+import org.hisp.dhis.dxf2.deprecated.tracker.importer.context.WorkContext;
+import org.hisp.dhis.trackedentitydatavalue.TrackedEntityDataValueChangeLogService;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/**
+ * @author Zubair Asghar
+ */
+@ExtendWith(MockitoExtension.class)
+class EventManagerTest {
+
+  @Mock EventPersistenceService eventPersistenceService;
+
+  @Mock Map<EventProcessorPhase, EventProcessorExecutor> executorsByPhase;
+
+  @Mock EventProcessorExecutor eventProcessorExecutor;
+
+  @Mock TrackedEntityDataValueChangeLogService trackedEntityDataValueChangeLogService;
+
+  @Mock List<Checker> checkersRunOnDelete;
+
+  @Captor ArgumentCaptor<EventProcessorPhase> eventProcessorPhaseArgumentCaptor;
+
+  @InjectMocks EventManager subject;
+
+  @Test
+  void updateEventDataValuesShouldTriggerUpdatePostProcessors() throws JsonProcessingException {
```

Comment:
we agreed on a naming convention for our tests: _shouldXXwhenYY_
for instance, it could be `shouldTriggerUpdatePostProcessorsWhenEventDataValuesUpdated`